### DISCUSSION TITLE
Who's hiring (July 2022)

### DISCUSSION THREAD

**Discussion by Nutlope** [vercel/next.js#38458] [Ideas]
# What is this?
New month, new post! This discussion is for companies looking for people with Next.js experience. Both permanent positions and contract work are welcome.

# Companies
Please use the following format followed by a brief description of the role:

> COMPANY_NAME | ROLE | LOCATION | Full-time | $100-150k base | LINK

Please only post if you personally are part of the hiring company—no recruiting firms or job boards. Only one post per company.

Good luck with recruiting!

## Sample Job Post

Here's a good example job post.

> Vercel | Senior Full-Stack Engineer | Remote (US, Canada, Europe, or Asia) | Full-time | $100-225k base | [Link](https://vercel.com/careers/senior-full-stack-engineer-amer)
>
> Vercel’s hiring across the board on various engineering teams and positions, but the above link is for our full-stack engineering position. Vercel is innovating on and for the web and at the edge. We build tools and workflows developers love and deliver outstanding performance for end users.
>
> I've been at the company for 2 years and still find plenty of opportunity for impact, and love the teammates I'm working with. Feel free to apply using the link above or reach out to me directly.

---

*If you're looking for last month's discussion, see #37379. At least one company has [hired someone](https://twitter.com/Nutlope/status/1475929984174895105) as a result of it!* 

If you hire someone through this job board, please reach out to me (hassan@vercel.com) and let me know so I can send you some swag :)

If you're a developer looking to post about themselves, use #38459.

**Comment by GYusufu** [comment]
Herff Jones | Sr Web Developer (Full Stack) | REMOTE (US) | Full-time/W2 | $100-150k base |  https://www.herffjones.com/about/careers/job-openings/?gnk=job&gni=8a78859f7e50c1e9017e6deab2991462

Herff Jones is hiring across the board on various engineering teams and positions, but the above link is for our full-stack web developer position. 

Feel free to apply using the link above or reach out to me directly concerning this role, another role on the site, or curiosity of the roles we have across our other business units under our parent company, Varsity Brands. Happy hunting, I look forward to speaking with you!

**Reply by webstackdev** [reply]
My class ring was from your company :)

**Reply by GYusufu** [reply]
> My class ring was from your company :)

I get that a lot! We're a large provider in the industry, glad to hear you know of us!

**Reply by HimusakiProgramming** [reply]
@GYusufu whats your Email?

**Reply by GYusufu** [reply]
> @GYusufu whats your Email?

Hi! My email is Gyusufu@VarsityBrands.com.

**Comment by seeARMS** [comment]
Paragraph | Full-stack Founding Engineer | Remote (US, Canada) | Full-time | $130k-180k base | [Link](https://paragraph.xyz/careers/founding-engineer)

Paragraph is a web3-native newsletter & publishing platform. We're building new ways for creators to engage & give ownership to their audience, create content together in decentralized writing collectives, and monetize via NFTs.

We recently raised a pre-seed from some notable crypto investors, we have >3k writers on the platform so far, and we're growing quickly.

**Comment by PeerRich** [comment]
| NAME | ROLE | LOCATION | TYPE | COMPENSATION | LINK |
|---|---|---|---|---|---|
| Cal.com, Inc | ~~Senior Full Stack Engineer~~ | ~~Remote~~ | ~~Full-time~~ | ~~$100k + series A equity~~ | [Apply](https://cal.com/jobs/) |

Cal.com is building the open-source scheduling infrastructure for the web. Anywhere from yoga teachers to governments use cal.com to schedule their appointments. Fully self-hosted / Whitelabel or hosted by us. https://github.com/calcom/cal.com was one of the fastest-growing commercial open source companies last year and won 2x Product of the Day, Product of Week, and Product of the Month on Product Hunt https://www.producthunt.com/products/cal-com

We're well-funded (32.4M over two rounds) by some silicon valley all-star investors. 

---

~~We're looking for a talented senior full-stack engineer who wants to work on complex timezone logic, availability blocking and calendar APIs~~

**Comment by iliassalvatore** [comment]
**[BEN ELLIOT $BET | Full Stack Developer | Europe | Freelance Part-time | [Link](https://benelliot.io/about)**

We're looking for an experienced Next/React dev to join our existing Back/Front-end team to finish this awesome project
with @PJaYOfficial and @michielvanderplas

Three.JS and WebGL parts that include an Interactive Avatar and 3D Graph have already been developed and are ready to be integrated, for the on-going development we’ve set up a Next.JS (react) framework, and Sanity Studio for back-end

🌎 **This project is supported by HTC and [Viverse](https://www.vive.com/us/viverse/) as part of the [ViveArts](https://www.vivearts.com/) artist partnerships! 🌎**

**🤔 Project**
- Ben Elliot Trading Page is a way to experience Ben’s avatar and buy into his NFT token
- The goal of this project is to elevate Ben Elliot’s online presence and experience, by launching a non-fungible token (asset) Ben is allowing his art network and fans to become part of a financially driven and incentivised ecosystem
- Creating the most advanced interactive and personalised experience on the web that centralises all the data surrounding Ben Elliot on the web into one accessible place
- Art and commerce merge into a single digital commodity; $BET, a fully fledged universe where Ben Elliot’s fans can engage, interact, become shareholders and earn rewards

**🚨 Required Skills**
- Full Stack Development and experience/interest in Three.JS, WebGL and 3D Animation
- Learn about real-time data streaming, displaying marketplace analytics, setting up user engagement streaks and most importantly, contribute to one of the most exciting and unique NFT projects in the Art and Crypto space to date
- By joining this project you’ll be working alongside top-tier Art Directors, Creative Web Designers, 3D Developers and become part of the Digital and Physical Art spaces, see our previous experience here: [Ilias Salvatore](https://www.linkedin.com/in/ilias-salvatore/) [PJ](https://www.linkedin.com/in/pjofficial/?originalSubdomain=nl) [Michiel van der Plas](https://www.linkedin.com/in/michielvanderplas/) and [Ben Elliot](https://www.porsche.com/dreams/global-gallery/ben-elliot)

**🚨 Duties**
- Collaborate on a daily basis with our Three.JS/WebGL developer
- Meticulously translate the UI/UX design wireframes to actual code for web and mobile first
- Develop new user-facing features
- Build reusable components and libraries for future use
- Optimize application for maximum speed and scalability
- Ensure the technical feasibility of UI/UX designs

**📈 Timeline/budget**
- Complete Development before end-July, end to end process no longer than 3/4 weeks
- Project budget available upon request

**🤔 Interested?**

- Reach out to directly via [iliassalvatore1@gmail.com](mailto:iliassalvatore1@gmail.com)] or Discord salvatore#3786
- Please familiarise yourself with our high level brief via the Notion link: https://iliassalvatore.notion.site/Back-Front-End-Development-Brief-a26bf3bf26af4a34a446ecd7f0844898

<img width="1279" alt="BET Preview 16 28 25" src="https://user-images.githubusercontent.com/95893817/178230495-0c333fec-593e-43a8-9547-440284a61523.png">
<img width="894" alt="Screen Shot 2022-07-11 at 11 13 30" src="https://user-images.githubusercontent.com/95893817/178230633-e3dc0a65-f5a0-495b-85ff-828e7cde0066.png">

**Comment by Dom1n1cr** [comment]
DAISY AI | Software Engineer (C++) | United Kingdom (Remote/Hybrid Greater Manchester) | Full-time | £40,000 - £70,000 base | [Link](https://daisy.ai/SoftwareEngineer.html)

Daisy AI Inc is an exciting early-stage startup founded by a group of construction industry experts and PhD level academics aiming to disrupt the construction industry with an offering of automated design tools based on artificial intelligence and machine learning methods. Part of Brainpool AI, our first product is an automated design tool for timber floors. We have successfully rolled out the product in the UK and we are now expanding into the residential construction market in North America. In the future we are aiming to extend the product portfolio to other construction components, based on market needs such as timber stairs, walls and roofs.

The primary responsibility for this role is to take ownership of the maintenance and development of a highly mature C/C++ code base. Ideal candidates will have 4 years+ in C/C++ or PhD with specific focused on using C/C++. In return we offer a basic salary of £40,000 - £70,000 and stock options in DAISY AI following a successful probation period. We are currently unable to offer visa sponsorship for this role.

If you are interested in applying or finding out more information you can hop on the link or drop me a message at [Dominic@brainpool.ai](mailto:Dominic@brainpool.ai)

**Comment by hwsimmons17** [comment]
Teleport | US - 100% remote | Senior Frontend Developer

Teleport is looking for a Senior Frontend Developer with a strong knowledge of TS paradigms, constructs, and idioms. As an application developer, you will lead us into the future by translating requirements into the design and development of customized systems in an agile environment. We are a web3 rideshare company built on the Solana blockchain. Our team is small but mighty. Our founders and earliest employees include experienced operators and engineers who have worked on distributed protocols, consumer products, games, and mobile applications. We are here because we want to see a world in which it is possible for everyone to participate in meaningful work and reap the rewards of co-ownership. Our investors include some of Uber’s earliest employees and the founders of billion dollar corporations. We recently raised capital and have enough in the bank to weather the current downturn. 

Salary: $120k - $180k - there is flexibility for perfect fit candidates
100% WFH
Minimum Total Experience: 10 years
Minimum Next.js Experience: 2 years

Additional benefits to be added over the coming months.

**Reply by hwsimmons17** [reply]
If you are interested in the role, please email me at hunter@dec.dev

**Comment by srindom** [comment]
Medusa | Frontend Engineer, Docs | Remote (US or Europe) | Full-time | $80-110k base | [Link](https://medusajs.notion.site/Frontend-Engineer-Docs-be28a5ed58444803b4936f7703bab617)

Medusa is looking for a Frontend Engineer with strong React experience to join our docs team. Medusa is the leading open-source e-commerce platform for Javascript developers. It provides the foundational building blocks for tomorrow’s e-commerce infrastructure by combining an open architecture with a headless e-commerce setup.

Medusa is among the fastest-growing projects on GitHub and is already being used by ambitious e-commerce companies - from global retailers to hyper-growth startups.

**Reply by ghost** [reply]
Hi, srindom.
As a senior full stack developer, I have expertise in building web pages in React using CSS framework(Tailwind CSS) and Material UI(UI component framework) and React's advanced features including Reusable components, Hooks, Context, Styled components, etc.

My previous works include followings, which are all full-stack & crypto projects.
 - https://app.yesports.gg/
 - https://moonstartoken.com
 - https://vigilantes.netlify.app/
 ...

I've always been focused on the clean and robust code, the strict deadline, and responsive and pixel perfect UX/UI.
My proficient skills, rich experience and positive attitude will make me a great fit for your job.

Thank you for taking your time and paying attention to me.
Kind regards,

**Reply by imabp** [reply]
Hi @srindom,  I am Abir. 
A **frontend engineer** and a **developer advocate**. I have realized the potential of MedusaJS, and would love to contribute towards it engineering and growth.

###  Work I am proud of

https://build.affinidi.com 
- Built the homepage, authentication flow, UI, and wallet creation UI.

https://app.codiga.io
-  Migrated components to Chakra UI from styled-components, and built on top of Codemirror.

### Tools and Framework Specilization

0. SASS, DOM Manipulation, Web APIs
1. Javascript + with Static Typing (Typescript)
2. React SPA , React-Query, Graphql, Storybook and Design Systems
3. NextJS (including usage of javascript ORMs and ODMst)
4. API Endpoint Testing following OWASP Industry standard security rules.
5. Packaging libraries, and necessary tooling to make it easy for developers.
6. NodeJS Server Architectures.


### Open Source and Advocacy Work

- Blogs
    - Apart from engineering, I add value to my surroundings, by documenting and helping others, hence defining a growth culture.
    - Technical Blogs: 
        - [Testing 101 in JS](https://javascript.plainenglish.io/testing-101-in-javascript-720c752ecfd5)
        - [Advanced Component Library using Chakra UI](https://www.codiga.io/blog/component-library-chakra-ui-react/)
    - Twitter Threads: 
        - I write twitter threads about different technologies internals, Check here I am explaining Context API in [here](https://twitter.com/imabptweets/status/1461005256817459202).
        - Apart from general work, I can explain powers of MedusaJS, to the people.

- Also Active in open source and developer communities
 
- Recognised as mentee  at AsyncAPI by Linux Foundation
     - Responsible for building a common error interface for org-wide tools, being
       consumed by walmart.
     - https://github.com/asyncapi/community/discussions/376#discussioncomment-2890658
     
- Major League Hacking(MLH) Fellow 

- Invited Speaker at Postman API Conference, Gold Microsoft Student Ambassador


> If all of this sounds interesting and promising, lets meet.
> Email: abir.pal899@gmail.com

**Comment by hect1c** [comment]
Palqee | Senior Frontend Engineer | Remote | Contract (anywhere), Full-time (UK, Brazil only)

Senior Frontend Engineer

Who we are 

Palqee is an investment backed start-up based out of London UK, with a solid product-market fit, impressive growth, clients in the UK & Brazil and an amazing team (https://palqee.com/). We're now looking for a Senior Front-end Engineer to join our superstar team and to lead, innovate, design, as well as play an active role in developing our application roadmap. 

What we are looking for

We are seeking an exceptional Senior Front-end engineer who wants to be part of bringing innovation, creativity, and security to the data privacy world. You are motivated in delivering well documented, clean, and efficient code with a passion and curiosity for the art of programming. You enjoy living by the code of your craft and developing mind-blowing solutions for complex problems. You are a true techie at heart. 

As one of the first key hires to grow our teach team, you will report directly to the CTO and  have influence and real impact in an early stage start-up on the expansion of our engineering team to build products and features that accelerate our traction.

You enjoy collaborating with different Team members and can turn ideas into tangible solutions but can also work independently with speed and efficiency. 

Responsibilities

- Provide engineering leadership with a focus on user interface design and development
- Design and implement user interfaces in collaboration with team of full stack engineers
- Experience working in agile environments that have embraced DevOps and Continuous Delivery best practices
- Developing automated tests, and deployment processes
- Insistent on recognising system imperfections and replace mistakes with rapid improvement

Skills

- 5+ years of related professional experience in a structured, process-driven environment
- 5+ years of experience coding HTML, CSS, Javascript and other UI technologies with a portfolio of your work
- Highly proficient developing in a Frontend library / ecosystem (React [preferable], VueJS, SolidJS, etc)
- Experience with GraphqQL
- Experience with a variety of JavaScript frameworks and libraries such as: Xstate, Apollo Client, NextJS, Typescript, Jest, Cypress, and Chai to name a few
- Enjoy working in a cross-functional team
- Must be responsive, detail oriented and have a strong sense of appropriate planning for meeting deadlines
- Strong sense of ownership and accountability

If interested please email me at [hartley@palqee.com](hartley@palqee.com)

**Comment by mattwoodco** [comment]
Underbelly | Full-Stack Engineer | Remote (US-only) | Full-time | $140-180k base | [Link](https://underbelly.breezy.hr/p/de74306131a1)

We are growing our Product engineering team at Underbelly, and looking for ambitious front-end and full-stack developers. We are a NextJS shop, and rely on modern tooling for the best developer experience. If you’re eager to work on creating best-in-class user interfaces and apps, please apply here: https://underbelly.breezy.hr/p/de74306131a1

![silicon15_12_wide-e759ee5cb01120bf93843294bea4737413764cdc](https://user-images.githubusercontent.com/566633/181382088-9606c84f-3f90-4e78-b7b8-0d76f617303d.jpg)

**Comment by Bartesque** [comment]
CONTRACT WORK | next.js, Prism, React,... | Remote (US, Canada, Europe, or Asia) | Part-time | Let me know your rate in DMs 

I'm looking for a dev to help me adapt the Platforms Kit to create MVP concepts.  So if you're familiar with it - fantastic. If not, experience working with multi-tenant platforms should help. 

Looking forward - thanks for any interest!!

**Reply by kevinorin** [reply]
Hello, I'm interested in learning more.

**Reply by Phoenix-Alpha** [reply]
Hey interested in this project. Please reach me out via alexjin0704@gmail.com

**Reply by Bartesque** [reply]
@kevinorin Super! Pls send a link to LinkedIn, or site or email where we can connect! @Phoenix-Alpha Super - reaching out now!

**Reply by asterism0115** [reply]
Hi, I am interested in this role, Please DM me via goldstar11520@gmail.com

**Reply by GreatWorldDev** [reply]
Hi, would love to discuss more the project. 
Please message me 
great.world.dev@gmail.com

**Reply by Alex-Garay** [reply]
Hello @Bartesque , I'm interested and would like to discuss the role further. Please email me at alex.garay94@gmail.com

**Comment by nirmpate** [comment]
HOWL | SENIOR SOFTWARE ENGINEER  | US REMOTE | Full-time | [LINK](https://grnh.se/16d9af072us)

About Howl 

Howl is the creator platform for social commerce. We build tools for brands and creators to sell more products across every social platform, together.

In the last 10 years, the $20bn influencer ecommerce market has become a critical part of the retail industry.  Social commerce is a new $1.2 trillion market opportunity ([link](https://www.accenture.com/us-en/insights/software-platforms/why-shopping-set-social-revolution?c=acn_glb_socialcommercebusinesswire_12755585&n=mrl_0122)) and requires new tools, protocols, and technologies.  

Howl allows creators and brands to run experiments on what products to talk about and how best to talk about them. Paired with data delivered at the speed of social, creator financing for products, and feedback loops delivered in a beautifully designed platform, Howl takes the guesswork out of selling products on social platforms.

Howl’s mission is to reduce barriers for creators and brands to build social commerce businesses.  We are a remote-first company with a diverse team that reflects our vision. Our beta launched in mid 2022 and powers social commerce for creators and brands across beauty, gaming, home, fashion, and consumer electronics for both industry leaders like Best Buy, Samsung, Target and cult favorites like Cariuma, Kosas, and Youth to the People. We are a Series A company backed by investors including HighlandX and NEA. 

For more information, visit [www.planethowl.com](http://www.planethowl.com/).

The position
Develop innovative algorithms for real time, high throughput, low latency marketplace system 
Successful experience designing and building large, scalable and complex distributed systems
Expertise in the Big Data Technology Stack
Ability to establish strong relationships with product and business stakeholders across the organization to co create objectives and facilitate their own cross-team scope and effort
Effective relationship builder throughout the engineering organization to refine initiatives and requirements, guide design and implementation, articulate outcome and impact
Creates & structures documentation for large code projects concisely & consistently
Ability to mentor teammates across teams based on experience & expertise
Relied upon across the engineering organization to keep complex projects on-track and sustainably implemented
Contribute to refinement and enrichment of development tools and processes
Ability to write persuasive analysis proposals to help make high-quality decisions about priorities among tech debt and tech strategy. Understands technical debt as a long-term budgeting and risk management tool; Can articulates when to incur more and when to pay it down.
Experience
A successful candidate will have substantial relevant experience in high-tech software development teams.
Experience in AdTech or Ecommerce. FinTech is OK too.
Experience in Big Data. Ideally, strong knowledge of ELT, Reverse ETL, DBT, Snowflake, Spark, 
Experience in Go and/or C++ and/or Python and/or Java programming and AWS
 

Education expectation
A bachelor’s or better degree in computer science or software engineering -or-
A bachelor’s or better degree in a STEM discipline plus substantial software engineering experience
 

Howl Tech Stack
One of the most frequent questions asked by candidates of Howl engineers is, “What is your tech stack?”  This is a brief overview of our tech stack.

Howl’s systems are implemented as a modern microservices architecture running on Linux servers hosted in AWS.  We use Kubernetes to manage our containers, we use Flask to construct our Web interfaces.  We build interactivity in our web interfaces using Nextjs.  We use Linux, and in particular Debian, Ubuntu, and Alpine distros.

Our favorite programming languages are Python 3, Scala, Go, Elixir, and, of course, TypeScript and JavaScript.  We stash our code in GitHub.

We test each language with an appropriate unit testing tool - JUnit, PyTest, ScalaTest, ExUnit, and Jasmine.  We use Jenkins and Github to run our builds and tests.

Check-ins against the main branch require a code review by another SWE.

We keep some data in Snowflake, some in Postgres, others in DynamoDB, and some, of course, in MySQL.  We build our product graph using Elasticache.  We are working to reduce the number of databases we rely on.

We stream data around using Kafka, Spark, and Storm.  Maybe we will prune this list sometime.

We are not hostile to commercial software - we use AirTable, DataDog, Fivetran, Jira, LaunchDarkly, Looker, Matomo, and many more.

**Comment by IraES0116** [comment]
**EQUUS SOFTWARE| SOFTWARE ENGINEER (FRONT END) | LISBON | Full-time | https://apply.workable.com/equus-software/j/32DB727635/**

As a Software Engineer, you will be building out key parts of our framework
that is used to build new products.


You will work with self-organizing development teams to improve the team,
process, product and user experience. Day-to-day, you will collaborate with
your team of developers, business analysts and product owners. You will also
work with subject matter experts from other teams as well as
implementation, support staff and sometimes Clients to produce a steady
stream of new features

Requirements
WHAT YOU BRING
3+ years proficiency with at least one object-oriented programming language (C#, Java, etc.)
Experience with **Javascript SPA like Angular/React, Angular 2, TypeScript, JQUERY, Twitter Bootstrap, JavaScript, CSS, HTML**
Understanding of complex concepts, such as recursion, reflection, multi-threaded processes, memory management
Proven record of leading and successful delivery of multiple projects
Experience working in both individual and team environments
NICE TO HAVE
**4+ years of experience operating in an agile environment (Scrum or Kanban)
Experience with ASP.NET MVC
Experience with Blazor**
Test Driven Development using a testing framework like MSTest or NUnit
Experience using an ORM framework like NHibernate or Entity
Framework
Object-oriented design
Database design
T-SQL
Code and database performance analysis
Secure coding practices
Troubleshooting complex problems
Visual Studio
SQL Server Management Studio
UI design
Benefits
Permanent Role – You will not start over when moving projects. We only hire passionate people who are looking for a long-term opportunity with a growing company.
We LOVE your ideas! Innovation Day, once a quarter you can share your fresh ideas.
Diversity/Inclusion - With employees from far-reaching countries across the globe, we've welcome unique mindsets and different backgrounds.
Culture – In Equus the work life balance is valued. For that reason, we try to organize team outings and team buildings activities regularly.
Salary & Benefits - We have highly competitive salaries with an exceptional benefit package, that includes medical coverage, PTO, flexible work hours, plus more.
Great people, great leaders. We only hire the BEST in the market so they can become the best professionals of industry! Our people are always available to help one another.
Yearly team events where you’ll be able to meet all of your colleagues face to face in one place.
